Intersect.Server/Entities/Entity.cs

```diff
@@ -2726,6 +2726,17 @@ public bool InRangeOf(Entity target, int range)
 
         public virtual void NotifySwarm(Entity attacker)
         {
+            if (!MapController.TryGetInstanceFromMap(MapId, MapInstanceId, out var instance))
+            {
+                return;
+            }
+
+            var entities = instance.GetEntities(true);
+
+            foreach (var entity in entities)
+            {
+                entity.OnNearEntityAttacked(this, attacker);
+            }
```

Comment:
Add to the for-loop before the OnSwarmAttacked call:

```csharp
            if (!CanSwarm)
            {
                continue;
            }
            
            if (!IsAllyOf(victim))
            {
                continue;
            }

            if (!InRangeOf(attacker, SightRange))
            {
                continue;
            }
```

Reply:
This requires that
- we add a `CanSwarm` virtual property to Entity, and the default value that it returns should be `false`
  - `Npc` needs an override that returns `Base.Swarm`
  - `Npc`'s `IsAllyOf` should have the `case Npc otherNpc` arm changed from `return Base == otherNpc.Base` to `return Base.Swarm && Base == otherNpc.Base` because if an NPC can't swarm then it really isn't an ally per the current code in `Npc`'s override for `NotifySwarm`
  - `Npc`'s override for `NotifySwarm` swarm should be deleted because all of its logic is actually part of the base `NotifySwarm` and `IsAllyOf` methods
- we add a `SightRange` virtual property to `Entity,` and the default value that it returns should be `999` (1K - 1, there's another special value at `9999` [10K-1] already)